### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by lucascomamala**
- Created model classes for all entities
- Set up associations between models
- Created methods for models:
  - A method that returns the 3 most recent posts for a given user.
  - A method that updates the posts counter for a user.
  - A method which returns the 5 most recent comments for a given post.
  - A method that updates the comments counter for a post.
  - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,15 @@
 class User < ApplicationRecord
+  has_many :posts, class_name: 'Post', foreign_key: 'author_id'
+  has_many :comments, class_name: 'Comment', foreign_key: 'user_id'
+  has_many :likes, class_name: 'Like', foreign_key: 'user_id'
+
+  # Returns the 5 most recent posts for the user
+  def recent_posts
+    posts.order(created_at: :desc).limit(5)
+  end
```
Comment (anagudelogu) [inline app/models/user.rb:9]: - [ ] According to the requirements:

image

Please correct the number of posts returned as it should be 3! 🤓 🔧

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,15 @@
...
+
+  # Method to update counters manually
+  def update_counters
+    update_attribute(:posts_counter, posts.all.count)
+  end
```
Comment (anagudelogu) [inline app/models/user.rb:14]: - [ ] **[OPTIONAL]** I think that having this "manually update" method is a repeating piece of code that you don't need as you're already defining the increment or decrement of this counter on Posts creation/deletion.

Kudos for using the hooks after_save and after_destroy! 🤓 🔧

File: app/models/post.rb [hunk 3]
```diff
@@ -1,2 +1,29 @@
...
+  # Returns the 5 most recent comments for the post
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  # Method to update counters manually
+  def update_counters
+    update_attribute(:comments_counter, comments.all.count)
+    update_attribute(:likes_counter, likes.all.count)
+  end
```
Comment (anagudelogu) [inline app/models/post.rb:18]: - [ ] **[OPTIONAL]** Similarly, I think that you don't need this "manually update" method as you're already defining the increment or decrement of these counters on Comment/Like creation/deletion.

Kudos for using the hooks after_save and after_destroy! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @lucascomamala, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

### README 📜 

- [ ] 🚧 You did a great job documenting your project! 💯 However please make sure that you have a LICENSE and that its respective link in the README is working, you can follow these steps from [GitHub's official docs](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository) to set up a LICENSE. Microverse suggests using MIT! 🤓 🔧

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
Hi @lucascomamala, 🤩🔥 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- [ ] Great job on the functionality for processing the data in models 🚀
- [ ] Kudos for using the hooks after_save and after_destroy!✨✨
- [ ] Professional README 🔥
- [ ] Followed Gitflow 🔥

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**